File: Java/PetShop.java
import java.util.ArrayList;

// Kelas utama PetShop
public class PetShop {
    private class Product {
        private String name;
        private String category;
        private double price;

        // Constructor
        public Product(String name, String category, double price) {
            this.name = name;
            this.category = category;
            this.price = price;
        }

        // Getter dan Setter
        public String getName() { return name; }
        public void setName(String name) { this.name = name; }

        public String getCategory() { return category; }
        public void setCategory(String category) { this.category = category; }

        public double getPrice() { return price; }
        public void setPrice(double price) { this.price = price; }
    }

    private ArrayList<Product> products = new ArrayList<>();

    public void addProduct(String name, String category, double price) {
        products.add(new Product(name, category, price));
    }

    public void displayProducts() {
        if (products.isEmpty()) {
            System.out.println("🚫 Tidak ada produk dalam daftar.");
            return;
        }
        System.out.println("\n=== Daftar Produk ===");
        System.out.println("+---------------------------------------------------+");
        System.out.println("| Nama Produk        | Kategori       | Harga       |");
        System.out.println("+---------------------------------------------------+");
        for (Product p : products) {
            System.out.printf("| %-18s | %-14s | Rp %.2f |\n",
                    p.getName(), p.getCategory(), p.getPrice());
        }
        System.out.println("+---------------------------------------------------+");
    }

    public void updateProduct(String oldName, String newName, String newCategory, double newPrice) {
        for (Product p : products) {
            if (p.getName().equalsIgnoreCase(oldName)) {
                p.setName(newName);
                p.setCategory(newCategory);
                p.setPrice(newPrice);
                System.out.println("✅ Produk berhasil diperbarui!");
                return;
            }
        }
        System.out.println("❌ Produk tidak ditemukan!");
    }

    public void deleteProduct(String name) {
        for (int i = 0; i < products.size(); i++) {
            if (products.get(i).getName().equalsIgnoreCase(name)) {
                products.remove(i);
                System.out.println("🗑 Produk berhasil dihapus!");
                return;
            }
        }
        System.out.println("❌ Produk tidak ditemukan!");
    }

    public void searchProduct(String name) {
        for (Product p : products) {
            if (p.getName().equalsIgnoreCase(name)) {
                System.out.println("\n🔎 Detail Produk:");
                System.out.println("Nama    : " + p.getName());
                System.out.println("Kategori: " + p.getCategory());
                System.out.println("Harga   : Rp " + p.getPrice());
                return;
            }
        }
        System.out.println("❌ Produk tidak ditemukan!");
    }
}

File: Python/petshop.py
# Kelas untuk merepresentasikan produk
class Product:
    def __init__(self, name, category, price):
        self._name = name
        self._category = category
        self._price = price

    # Getter dan Setter untuk nama produk
    @property
    def name(self):
        return self._name
    @name.setter
    def name(self, new_name):
        self._name = new_name

    # Getter dan Setter untuk kategori produk
    @property
    def category(self):
        return self._category
    @category.setter
    def category(self, new_category):
        self._category = new_category

    # Getter dan Setter untuk harga produk
    @property
    def price(self):
        return self._price
    @price.setter
    def price(self, new_price):
        self._price = new_price

# Kelas utama PetShop untuk mengelola produk
class PetShop:
    def __init__(self):
        # Constructor untuk inisialisasi daftar produk
        self.products = []

    def add_product(self, name, category, price):
        # Menambahkan produk baru ke dalam daftar
        self.products.append(Product(name, category, price))
        print("✅ Produk berhasil ditambahkan!")

    def display_products(self):
        # Menampilkan daftar produk dalam bentuk tabel
        if not self.products:
            print("🚫 Tidak ada produk dalam daftar.")
            return
        
        print("\n=== Daftar Produk ===")
        print("+-----------------------------------+")
        print("| Nama Produk    | Kategori | Harga |")
        print("+-----------------------------------+")
        for p in self.products:
            print(f"| {p.name:<14} | {p.category:<8} | Rp {p.price:.2f} |")
        print("+-----------------------------------+")

    def update_product(self, old_name, new_name, new_category, new_price):
        # Memperbarui produk berdasarkan nama produk
        for p in self.products:
            if p.name.lower() == old_name.lower():
                p.name = new_name
                p.category = new_category
                p.price = new_price
                print("✅ Produk berhasil diperbarui!")
                return
        print("❌ Produk tidak ditemukan!")

    def delete_product(self, name):
        # Menghapus produk berdasarkan nama produk
        for i, p in enumerate(self.products):
            if p.name.lower() == name.lower():
                del self.products[i]
                print("🗑 Produk berhasil dihapus!")
                return
        print("❌ Produk tidak ditemukan!")

    def search_product(self, name):
        # Mencari produk berdasarkan nama
        for p in self.products:
            if p.name.lower() == name.lower():
                print("\n🔎 Detail Produk:")
                print(f"Nama    : {p.name}")
                print(f"Kategori: {p.category}")
                print(f"Harga   : Rp {p.price:.2f}")
                return
        print("❌ Produk tidak ditemukan!")
